src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/WheelchairCostCalculator.java

```diff
@@ -0,0 +1,99 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.cost;
+
+import static org.opentripplanner.model.WheelChairBoarding.NOT_POSSIBLE;
+import static org.opentripplanner.model.WheelChairBoarding.NO_INFORMATION;
+import static org.opentripplanner.model.WheelChairBoarding.POSSIBLE;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nonnull;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.TripSchedule;
+import org.opentripplanner.routing.api.request.WheelchairAccessibilityRequest;
+import org.opentripplanner.transit.raptor.api.transit.CostCalculator;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransfer;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransferConstraint;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTripSchedule;
+
+public class WheelchairCostCalculator implements CostCalculator {
+
+  private final CostCalculator delegate;
+  private final int[] wheelchairBoardingCost;
+
+  public WheelchairCostCalculator(
+    @Nonnull CostCalculator delegate,
+    @Nonnull WheelchairAccessibilityRequest requirements
+  ) {
+    // assign the costs for boarding a trip with the following accessibility values
+    wheelchairBoardingCost = new int[3];
+    wheelchairBoardingCost[POSSIBLE.ordinal()] = 0;
+    wheelchairBoardingCost[NO_INFORMATION.ordinal()] =
+      RaptorCostConverter.toRaptorCost(requirements.trips().unknownCost());
+    wheelchairBoardingCost[NOT_POSSIBLE.ordinal()] =
+      RaptorCostConverter.toRaptorCost(requirements.trips().inaccessibleCost());
+
+    this.delegate = delegate;
+  }
+
+  @Override
+  public int boardingCost(
+    boolean firstBoarding,
+    int prevArrivalTime,
+    int boardStop,
+    int boardTime,
+    RaptorTripSchedule trip,
+    RaptorTransferConstraint transferConstraints
+  ) {
+    int defaultCost = delegate.boardingCost(
+      firstBoarding,
+      prevArrivalTime,
+      boardStop,
+      boardTime,
+      trip,
+      transferConstraints
+    );
+    // this is an unsafe cast but is ok because the trip is guaranteed to be of type TripSchedule
+    var tripSchedule = (TripSchedule) trip;
+    int index = tripSchedule.getOriginalTripTimes().getTrip().getWheelchairBoarding().ordinal();
```

Comment:
This is going to be slow! You can add `getWheelchairBoarding()` to the `TripSchedule` interface as a default method and cash the value of the enum in `TripScheduleWithOffset` - so the Raptor execution do not have to access tripTimes. You may do this in a separate PR later - on just leave it as is. I am changing this in the new model.

Reply:
Ok, in a previous version I did exactly that but after some discussion us two decided that we will keep it the way it is.

Reply:
Since it made writing the tests easier I ended up adding a method to `TripSchedule`.